### PR TITLE
sys-apps/rng-tools: open entropy src with O_NOCTTY flag wrt bug #556456.

### PR DESCRIPTION
**Pull request by gktrk**
This revision patches the source file 'rngd_entsource.c', adding O_NOCTTY
flag to the open() call that opens the entropy source for rngd.

Package-Manager: portage-2.2.20.1

### CODE REVIEW

**Comment by mgorny** [comment]
@gentoo/base-system @gentoo/proxy-maint @idella

File: sys-apps/rng-tools/files/rng-tools-5-fix-noctty.patch [hunk 1]
```diff
@@ -0,0 +1,15 @@
+https://bugs.gentoo.org/556456
+
```
Comment (vapier) [inline sys-apps/rng-tools/files/rng-tools-5-fix-noctty.patch]: patch needs a description of what it's doing and why (and example).  please see: https://dev.gentoo.org/~vapier/clean-patches

File: sys-apps/rng-tools/files/rng-tools-5-fix-noctty.patch [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+patch by Gokturk Yuksek <gokturk@binghamton.edu>
+
+--- rngd_entsource.c.orig	2015-09-14 14:00:40.984919633 -0400
```
Comment (vapier) [inline sys-apps/rng-tools/files/rng-tools-5-fix-noctty.patch]: please clean up the metadata -- see aforementioned URL

**Comment by vapier** [comment]
wrt to the commit message, the "wrt bug" part is redundant.  just putting "#12345" at the end is enough.

File: sys-apps/rng-tools/rng-tools-5-r2.ebuild [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+KEYWORDS="~alpha ~amd64 ~arm ~ia64 ~ppc ~x86"
+IUSE="selinux"
+
+DEPEND="dev-libs/libgcrypt:0
+	dev-libs/libgpg-error"
+RDEPEND="${DEPEND}
+	selinux? ( sec-policy/selinux-rngd )"
+
+src_prepare() {
+	echo 'bin_PROGRAMS = randstat' >> contrib/Makefile.am
```
Comment (mgorny) [inline sys-apps/rng-tools/rng-tools-5-r2.ebuild]: ||die

**Comment by idella** [comment]
gokturk, mgorny is quite right re adding " || die" to the line of echo 'bin_PROGRAMS however
use helper func einfo simply calls echo and doesn't need the call of || die

File: sys-apps/rng-tools/rng-tools-5-r2.ebuild (hunk 3 above)
Comment (gktrk) [inline sys-apps/rng-tools/rng-tools-5-r2.ebuild]: This was part of the original ebuild. I will add die as a separate commit. Are you ok with it being a part of this PR?

Comment (mgorny) [inline sys-apps/rng-tools/rng-tools-5-r2.ebuild]: Yes. It's fine to add extra fixes.

**Comment by mgorny** [comment]
Just to be clear, I encourage die when directing output to a file since opening/writing this file could fail.

**Comment by gktrk** [comment]
I would like to close this bug along with #481254 in this PR to avoid a revbump. Is that ok if I include those fixes in the next push too?

**Comment by mgorny** [comment]
Yes, it's ok to include bugfixes in version bumps.

**Comment by gktrk** [comment]
This new push includes fixes for #481254 and #556456, as well as changes requested by the devs here.

**Comment by mgorny** [comment]
@vapier, anything left to do here?

**Pull request merged**